apps/studio/components/interfaces/RoleImpersonationSelector/UserImpersonationSelector.tsx

```diff
@@ -0,0 +1,202 @@
+
+  const impersonatingUser =
+    state.role?.type === 'postgrest' && state.role.role === 'authenticated' && state.role.user
+
+  function impersonateUser(user: User) {
+    state.setRole({
+      type: 'postgrest',
+      role: 'authenticated',
+      user,
+    })
+  }
+
+  function stopImpersonating() {
+    state.setRole(undefined)
+  }
+
+  return (
+    <div className="flex flex-col gap-1">
+      <h2 className="text-foreground text-base">
+        {impersonatingUser
+          ? `Impersonating ${getDisplayName(
+              impersonatingUser,
+              impersonatingUser.email ??
+                impersonatingUser.phone ??
+                impersonatingUser.id ??
+                'Unknown'
+            )}`
+          : 'Impersonate a User'}
+      </h2>
+      <p className="text-sm text-foreground-light max-w-md">
+        {!impersonatingUser
+          ? "Select a user to respect your database's Row-Level Security policies for that particular user."
+          : "Results will respect your database's Row-Level Security policies for this user."}
+      </p>
+
+      {!impersonatingUser ? (
+        <div className="flex flex-col gap-2 mt-4">
+          <Input
+            className="table-editor-search border-none"
+            icon={
+              isSearching ? (
+                <Loader2
+                  className="animate-spin text-foreground-lighter"
+                  size={16}
+                  strokeWidth={1.5}
+                />
+              ) : (
+                <Search className="text-foreground-lighter" size={16} strokeWidth={1.5} />
+              )
+            }
+            placeholder="Search for a user.."
+            onChange={(e) => setSearchText(e.target.value.trim())}
+            value={searchText}
+            size="small"
+            actions={
+              searchText && (
+                <Button size="tiny" type="text" onClick={() => setSearchText('')}>
+                  <X size={12} strokeWidth={2} />
+                </Button>
+              )
+            }
+          />
+
+          {isLoading && (
+            <div className="flex flex-col gap-2 items-center justify-center h-24">
+              <Loader2 className="animate-spin" size={24} />
+              <span>Loading users...</span>
+            </div>
+          )}
+
+          {isError && <AlertError error={error} subject="Failed to retrieve users" />}
+
+          {isSuccess &&
+            (data.users.length > 0 ? (
+              <ul className="divide-y max-h-[192px] overflow-y-scroll" role="list">
+                {data.users.map((user) => (
+                  <li key={user.id} role="listitem">
+                    <UserRow user={user} onClick={impersonateUser} />
+                  </li>
+                ))}
+              </ul>
+            ) : (
+              <div className="flex flex-col gap-2 items-center justify-center h-24">
+                <p className="text-foreground-light" role="status">
```

Comment:
```suggestion
                <p className="text-foreground-light text-xs" role="status">
```

Reply:
actually could we use the same UI for e.g when we're searching in database/indexes?

No results found
Your search for "{search}" did not return any results

probably a good time to make this a reusable component actually - just realised its repeated in multiple places if you cmd shift f "No results"